### PR TITLE
fix(driver): lv_x11_input.c fixes

### PR DESCRIPTION
**Pull request by liamHowatt**
### Description of the feature or fix

A regression since #5767. lv_x11_input.c was missed in the refactor. Update the includes.

Also remove unnecessary `LV_ASSERT_OBJ`s that will not compile if `LV_USE_ASSERT_OBJ` is `1`.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by XuNeo — APPROVED** [review]

File: src/drivers/x11/lv_x11_input.c [hunk 1]
```diff
@@ -10,12 +10,11 @@
 
 #if LV_USE_X11
 
+#include <string.h>
 #include <stdbool.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
-#include "../../stdlib/lv_string.h"
```
Comment (kisvegabor) [inline src/drivers/x11/lv_x11_input.c:16]: I think we should keep `lv_string.h` and use `lv_memmove` [here](https://github.com/lvgl/lvgl/blob/master/src/drivers/x11/lv_x11_input.c#L229).

Comment (XuNeo) [reply src/drivers/x11/lv_x11_input.c:16]: x11 lib probably needs those standard headers anyway, so I think it's OK to use `memmove` etc. in drivers/.

Comment (liamHowatt) [reply src/drivers/x11/lv_x11_input.c:16]: I looked at what was done for the other platform-specific drivers. Standard headers seem to be used intentionally.
https://github.com/lvgl/lvgl/pull/5767/files#diff-f987953b6b5639df99192994498283ec69629b506874914605a14fd7afd12728

Comment (kisvegabor) [reply src/drivers/x11/lv_x11_input.c:16]: Okay, let's keep the suggested changes.

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**